builtin/providers/openstack/resource_openstack_compute_instance_v2.go

```diff
@@ -659,3 +671,36 @@ func getFloatingIPs(networkingClient *gophercloud.ServiceClient) ([]floatingips.
 	}
 	return ips, nil
 }
+
+func getImageID(client *gophercloud.ServiceClient, d *schema.ResourceData) (string, error) {
+	imageID := d.Get("image_id").(string)
+	imageName := d.Get("image_name").(string)
+	if imageID == "" {
+		pager := images.ListDetail(client, nil)
```

Comment:
You should give query options to fetch only image(s) with the desired name. https://github.com/rackspace/gophercloud/blob/master/openstack/compute/v2/images/requests.go#L25